addons/mail/static/src/js/views/activity/activity_renderer.js

```diff
@@ -78,30 +121,60 @@ var ActivityRenderer = AbstractRenderer.extend({
      */
     _renderRow: function (data) {
         var self = this;
-        var res_id = data[0];
-        var name = data[1];
-        var $nameTD = $('<td>')
-            .addClass("o_res_name_cell")
-            .html(name)
-            .data('res-id', res_id);
+        var res_id = data.id;
+        if (data.currency_id) {
+            data.subtitle_currency = field_utils.format.monetary(data.subtitle, {}, {currency_id: data.currency_id});
+        }
+        var $nameTR = $(QWeb.render('mail.ActivityViewRowSidebar', {data: data}));
         var $cells = _.map(this.state.data.activity_types, function (node) {
-            var $td = $('<td>').addClass("o_activity_summary_cell");
             var activity_type_id = node[0];
             var activity_group = self.state.data.grouped_activities[res_id][activity_type_id];
             activity_group = activity_group || {count: 0, ids: [], state: false};
+
+            var $td = $(QWeb.render('mail.ActivityViewRow', {
+                data: data,
+                activity_group: activity_group,
+                activity_type_id: activity_type_id,
+            }));
             if (activity_group.state) {
-                $td.addClass(activity_group.state);
+                var record = self._getkanbanActivityData(activity_group, res_id);
```

Comment:
Here the logic below the line is an old logic but the diff generated because we've made a function for some common part and added some conditions to implement a kanban styled card. So basically first 'mail.ActivityViewRow' template will render and it'll return a TD jquery element then the 'kanbanActivity' widget appending to the TD and then to replace a clock icon by the closest deadline, the 'date' will append to the widget template instead of clock-icon.